src/processor/graph-update-completion-monitor.service.ts

```diff
@@ -0,0 +1,119 @@
+import { MILLISECONDS_PER_SECOND } from 'time-constants';
+import { BlockchainService } from '#app/blockchain/blockchain.service';
+import { BlockchainConstants } from '#app/blockchain/blockchain-constants';
+import { BlockchainScannerService } from '#app/blockchain-scanner.service';
+import { SchedulerRegistry } from '@nestjs/schedule';
+import { BlockHash } from '@polkadot/types/interfaces';
+import { HexString } from '@polkadot/util/types';
+import { ReconnectionCacheMgrService } from '#app/cache/reconnection-cache-mgr.service';
+
+@Injectable()
+export class GraphUpdateCompletionMonitorService extends BlockchainScannerService implements OnApplicationBootstrap, OnApplicationShutdown {
+  async onApplicationBootstrap() {
+    this.schedulerRegistry.addInterval(
+      `${this.constructor.name}:blockchainScan`,
+      setInterval(() => this.scan(), BlockchainConstants.SECONDS_PER_BLOCK * MILLISECONDS_PER_SECOND),
+    );
+
+    await super.onApplicationBootstrap();
+  }
+
+  async onApplicationShutdown(signal?: string | undefined) {
+    if (this.schedulerRegistry.doesExist('interval', `${this.constructor.name}:blockchainScan`)) {
+      this.schedulerRegistry.deleteInterval(`${this.constructor.name}:blockchainScan`);
+    }
+
+    super.onApplicationShutdown(signal);
+  }
+
+  constructor(
+    cacheManager: ReconnectionCacheMgrService,
+    private readonly schedulerRegistry: SchedulerRegistry,
+    blockchainService: BlockchainService,
+    private readonly cacheService: ReconnectionCacheMgrService,
+  ) {
+    super(cacheManager, blockchainService, new Logger(GraphUpdateCompletionMonitorService.name));
+  }
+
+  async processCurrentBlock(currentBlockHash: BlockHash, currentBlockNumber: number): Promise<void> {
+    // Get set of tx hashes to monitor from cache
+    let pendingTxns = await this.cacheService.getAllPendingTxns();
+
+    const block = await this.blockchainService.getBlock(currentBlockHash);
+    const extrinsicIndices: [HexString, number][] = [];
+    block.block.extrinsics.forEach((extrinsic, index) => {
+      if (Object.keys(pendingTxns).some((txHash) => txHash === extrinsic.hash.toHex())) {
+        extrinsicIndices.push([extrinsic.hash.toHex(), index]);
+      }
+    });
+
+    if (extrinsicIndices.length > 0) {
+      const at = await this.blockchainService.apiPromise.at(currentBlockHash);
+      const epoch = (await at.query.capacity.currentEpoch()).toNumber();
+      let totalCapacityWithdrawn: bigint = 0n;
+      const events = (await at.query.system.events()).filter(({ phase }) => phase.isApplyExtrinsic && extrinsicIndices.some((index) => phase.asApplyExtrinsic.eq(index)));
+      const capacityAmounts: bigint[] = events
+        .filter(({ event }) => at.events.capacity.CapacityWithdrawn.is(event))
+        .map(({ event }) => (event as unknown as any).data.amount.toBigInt());
+      totalCapacityWithdrawn = capacityAmounts.reduce((prev, curr) => prev + curr, totalCapacityWithdrawn);
```

Comment:
A bit of code golf to clean up the `totalCapacityWithdrawn`

```suggestion
      const events = (await at.query.system.events()).filter(({ phase }) => phase.isApplyExtrinsic && extrinsicIndices.some((index) => phase.asApplyExtrinsic.eq(index)));
      
      const totalCapacityWithdrawn: bigint = events
        .filter(({ event }) => at.events.capacity.CapacityWithdrawn.is(event))
        .reduce((sum, { event }) => (event as unknown as any).data.amount.toBigInt() + sum, 0n);
```